src/lib/field-next/field-foundation.ts

```diff
@@ -0,0 +1,290 @@
+import { ICustomElementFoundation } from '@tylertech/forge-core';
+import { IFieldAdapter } from './field-adapter';
+import { FieldDensity, FieldLabelAlignment, FieldLabelPosition, FieldShape, FieldSlot, FieldSupportTextInset, FieldTheme, FieldVariant, FIELD_CONSTANTS } from './field-constants';
+
+export interface IFieldFoundation extends ICustomElementFoundation {
+  labelPosition: FieldLabelPosition;
+  labelAlignment: FieldLabelAlignment;
+  floatLabel: boolean;
+  invalid: boolean;
+  required: boolean;
+  optional: boolean;
+  disabled: boolean;
+  variant: FieldVariant;
+  theme: FieldTheme;
+  shape: FieldShape;
+  density: FieldDensity;
+  dense: boolean;
+  popoverIcon: boolean;
+  popoverExpanded: boolean;
+  multiline: boolean;
+  supportTextInset: FieldSupportTextInset;
+  initialize(): void;
+  floatLabelWithoutAnimation(value: boolean): void;
+}
+
+export class FieldFoundation implements IFieldFoundation {
+  private _labelPosition: FieldLabelPosition = FIELD_CONSTANTS.defaults.DEFAULT_LABEL_POSITION;
+  private _labelAlignment: FieldLabelAlignment = FIELD_CONSTANTS.defaults.DEFAULT_LABEL_ALIGNMENT;
+  private _floatLabel = false;
+  private _invalid = false;
+  private _required = false;
+  private _optional = false;
+  private _disabled = false;
+  private _variant: FieldVariant = FIELD_CONSTANTS.defaults.DEFAULT_VARIANT;
+  private _theme: FieldTheme = FIELD_CONSTANTS.defaults.DEFAULT_THEME;
+  private _shape: FieldShape = FIELD_CONSTANTS.defaults.DEFAULT_SHAPE;
+  private _density: FieldDensity = FIELD_CONSTANTS.defaults.DEFAULT_DENSITY;
+  private _dense = false;
+  private _popoverIcon = false;
+  private _popoverExpanded = false;
+  private _multiline = false;
+  private _supportTextInset: FieldSupportTextInset = FIELD_CONSTANTS.defaults.DEFAULT_SUPPORT_TEXT_INSET;
+
+  private _labelSlotListener: () => void;
+  private _startSlotListener: () => void;
+  private _endSlotListener: () => void;
+  private _accessorySlotListener: () => void;
+  private _supportStartSlotListener: () => void;
+  private _supportEndSlotListener: () => void;
+  private _popoverIconClickListener: () => void;
+
+  constructor(private _adapter: IFieldAdapter) {
+    this._labelSlotListener = () => this._onSlotChange('label');
+    this._startSlotListener = () => this._onSlotChange('start');
+    this._endSlotListener = () => this._onSlotChange('end');
+    this._accessorySlotListener = () => this._onSlotChange('accessory');
+    this._supportStartSlotListener = () => this._onSlotChange('support-start');
+    this._supportEndSlotListener = () => this._onSlotChange('support-end');
+    this._popoverIconClickListener = () => this._onPopoverIconClick();
+  }
```

Comment:
You could probably combine the declarations and initialization using `.bind(this, <type>)` to reduce some code:
```ts
private _labelSlotListener: () => void = this._onSlotChange.bind(this, 'label');
```